src/indent.cpp

```diff
@@ -1792,11 +1792,12 @@ void indent_text(void)
                         && frm.at(idx).type != CT_ANGLE_OPEN
                         && frm.at(idx).type != CT_CASE
                         && frm.at(idx).type != CT_MEMBER
+                        && frm.at(idx).type != CT_LAMBDA
                         && frm.at(idx).type != CT_ASSIGN_NL)
                      || are_chunks_in_same_line(frm.at(idx).pc, frm.top().pc))
                   && (  frm.at(idx).type != CT_CLASS_COLON
                      && frm.at(idx).type != CT_CONSTR_COLON
-                     && frm.at(idx).type != CT_LAMBDA))
+                     && !(frm.at(idx).type == CT_LAMBDA && chunk_get_prev_nc(frm.at(idx).pc)->type == CT_NEWLINE)))
```

Comment:
Why do you put the '!' in front instead of resolving the expression? 

It is not improving performance, I tested this for a simpler example with [godbolt](https://gcc.godbolt.org/) and it produces the same assembly as the resolved form.
<details>

```cpp
struct ZZ
{
    int i = 9;
    bool f = false;
};

bool z(int i)
{
    ZZ z;
    z.i = i * 0.5;

    return (z.i * 3) == 15;
}


bool f ( bool b, bool c, bool d )
{
    bool a = false || d;

    if (!(b && z(1 * c)))
    {
        return a;
    }
    return false;

}

bool ff ( bool b, bool c, bool d )
{
    bool a = false || d;

    if (!b || !z(1 * c))
    {
        return a;
    }
    return false;
}

```
</details>